src/net/neighbors.rs

```diff
@@ -3390,6 +3394,103 @@ mod test {
         })
     }
 
+    #[test]
+    #[ignore]
+    fn test_step_walk_1_neighbor_bad_epoch() {
+        with_timeout(600, || {
+            let mut peer_1_config = TestPeerConfig::from_port(31998);
+            let mut peer_2_config = TestPeerConfig::from_port(31990);
+
+            peer_1_config.connection_opts.walk_retry_count = 10;
+            peer_2_config.connection_opts.walk_retry_count = 10;
+            peer_1_config.connection_opts.walk_interval = 1;
+            peer_2_config.connection_opts.walk_interval = 1;
+
+            // peer 1 thinks its always epoch 2.0
+            peer_1_config.peer_version = 0x18000000;
+            peer_1_config.epochs = Some(vec![StacksEpoch {
```

Comment:
What kind of testing do we need to do? or can we do? in order to fully understand this feature?

Reply:
The unit tests added here should be sufficient.  If you want, you can leave a 2.0 testnet node running and verify that it loses all its neighbor connections on the epoch transition

Reply:
I can walk through the test if there is a possible one. Like if we can construct a test with the private testnet or something.